ClientApp/viking-vault/src/components/TopBar.tsx

```diff
@@ -1,13 +1,58 @@
 import React from 'react'
 import './styles.css'
 import logo from './images/logo.png'
+import { Link } from 'react-router-dom';
+import {variables} from "../ConstantVariables";
 
-class TopBar extends React.Component{
+const baseUrl = variables.baseUrl;
+
+interface IState {
+    isAdmin: boolean;
+}
+
+class TopBar extends React.Component<any, IState>{
+    constructor(props:any) {
+        super(props);
+        this.state = {
+            isAdmin: false
+        }
+    }
+
+    private checkIfUserIsAdmin = () => {
+        let token = sessionStorage.getItem("Authentication-Token");
+        if(token === null) {
+            token = "";
+        }
+        fetch(baseUrl+"admin", {
+            method: "POST",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token' : token.toString()
+            }
+        }).then(response => response.json())
+        .then(result => {
+            if(result === true) {
```

Comment:
Why do you need the if/else blocks here, couldn't you simple say `this.setState({ isAdmin = result });`